### PR TITLE
Rewrite REPL, add graalvm building, move to drone

### PR DESCRIPTION
**Pull request by andimiller**
* Swapped travis out for drone since it was an easier way to add the graal step
* The new REPL uses jline3 to unify jline versions, and idmltutor has been bumped to match
* The new REPL has a clear separation between the pure code and impure code
* Proguard has been added to reduce the jar size a bit for the tool
* Graal is now used to make a native binary of the tool

### CODE REVIEW

File: tool/src/graal/resource-config.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (jcalonso) [inline tool/src/graal/resource-config.json]: How is this used?

Comment (andimiller) [reply tool/src/graal/resource-config.json]: this is fed into the graal build, you can see it being used in the drone config

File: repl/build.sbt [hunk 2]
```diff
@@ -1,12 +1,50 @@
...
+    |    public static ** valueOf(java.lang.String);
+    |}""".stripMargin
+)
+
+Proguard / proguardInputs := (dependencyClasspath in Compile).value.files
+Proguard / proguardMergedInputs ++= ProguardOptions.noFilter((packageBin in Compile).value)
+Proguard / proguard / javaOptions := Seq("-Xmx2g")
+Proguard / proguardOutputs := Seq(new File(target.value, "repl-proguard.jar"))
+
+proguardOptions in Proguard += ProguardOptions.keepMain("io.idmlrepl.Main")
```
Comment (nicktelford) [inline repl/build.sbt]: Why is this not:
`Proguard / proguardOptions`? Does this style syntax even work in SBT anymore?

Comment (andimiller) [reply repl/build.sbt]: it still works on 1.2.8, it may need to change when I upgrade

**Pull request merged**